### PR TITLE
fix(docs-infra): typography, layout and UI polish

### PR DESCRIPTION
**Pull request by sjtrimble**
**As discussed with @gkalpak - splitting existing PR #36045 into smaller bit sized PRs**

- Sidenav background to better distinguish from content
- Header styles updated
- General typography styles updated (removed letter-spacing, increased font size and updated line height)
- Prominent h2 styles with line to segment important sections more easily
- Vertical align headerin glink icons (on hover)
- Prominent link hover underline (more accessible)
- Max width on content paragraph text to improve legibility (best practices)
- Centered layout for better legibility and focus on content
- API pages override to maintain current styles

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?
N/A

Issue Number: N/A

## What is the new behavior?
UI updates with a focus on improved legibility

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: aio/src/styles/0-base/_typography.scss [hunk 1]
```diff
@@ -6,59 +6,73 @@ body {
...
   color: $darkgray;
-  @include font-size(14);
+  @include font-size(16);
   -webkit-font-smoothing: antialiased;
   -moz-osx-font-smoothing: grayscale;
 }
 
 h1, h2, h3, h4, h5, h6 {
   color: $deepgray;
+  margin-bottom: 0;
```
Comment (gkalpak) [inline aio/src/styles/0-base/_typography.scss:16]: Why this change? It causes the package name be too close to the symbols in the API list, which is less readable imo:

_**Before:**_
api-h2-before

_**After:**_
api-h2-after

File: aio/src/styles/0-base/_typography.scss [hunk 2]
```diff
@@ -6,59 +6,73 @@ body {
...
+}
+
+aio-shell {
+  &.page-resources, &.page-events, &.page-features, &.page-presskit, &.page-contribute, &.page-about, &.page-api {
+    main h2:before {
+      display: none;
+      border: 0;
+      visibility: hidden;
+    }
+  }
```
Comment (gkalpak) [inline aio/src/styles/0-base/_typography.scss:51]: I think it is better nest the special styles inside the `h2:before` block:

```suggestion

    .page-resources &, .page-events &, .page-features &, .page-presskit &, .page-contribute &, .page-about &, .page-api & {
      display: none;
      border: 0;
      visibility: hidden;
    }
  }
```

File: aio/src/styles/0-base/_typography.scss [hunk 3]
```diff
@@ -87,20 +101,17 @@ ol,
...
 input,
 a {
-  @include font-size(14);
-  @include line-height(24);
-  @include letter-spacing(0.3);
+  @include font-size(16);
+  @include line-height(32);
   font-weight: 400;
   color: $darkgray;
+  max-width: 50em;
```
Comment (gkalpak) [inline aio/src/styles/0-base/_typography.scss:108]: Why is this needed? Isn't setting the `.sidenav-content` max-width enough?

File: aio/src/styles/0-base/_typography.scss [hunk 4]
```diff
@@ -188,8 +199,9 @@ code {
 
 .sidenav-content a {
   color: $blue;
+
   &:hover {
-    color: $mediumgray;
+    text-decoration: underline;
```
Comment (gkalpak) [inline aio/src/styles/0-base/_typography.scss:204]: :100:

File: aio/src/styles/0-base/_typography.scss [hunk 5]
```diff
@@ -214,7 +225,7 @@ code {
         margin-left: 2px;
         position: relative;
         @include line-height(24);
-        vertical-align: middle;
+        vertical-align: bottom;
```
Comment (gkalpak) [inline aio/src/styles/0-base/_typography.scss:228]: This definitely doesn't llok good on my screen :grin:
external-links-alignment

Does `bottom` work better for you?

**Review by gkalpak — CHANGES_REQUESTED** [review]
I like some of the changes while not a big fan of others, but I think this is a positive change overall :smiley:

Besides the comments I left, I also noticed some other issues:

- The commit message mentions `Vertical align headerin glink icons (on hover)`. Apart from the obvious typo, I could find which changes this line refers to. Could it be that this is not included in this PR (or did I miss it)?

- Now that headings are bigger, the [current style][1] of `vertical-align: text-top` for heading links doesn't work well for me. `middle` works much better. How does `middle` look for you?

- Now that the font is bigger, it causes long API symbol names to be truncated:
    api-truncated

- For some reason, the footer alignment is broken:
    footer-alignment


[1]: https://github.com/angular/angular/blob/c349f7ed68b31d6f268ae5858803e0d073396b41/aio/src/styles/2-modules/_heading-anchors.scss#L11

**Review by Splaktar — CHANGES_REQUESTED** [review]
I like the change to make the docs sections made for reading limited to `800px` of width. Very long lines of text are harder for people to read.

However I think that pages like the Resource Listing, Events, and Presskit that are more for displaying a different kind of content should be allowed to be wider and use a different layout, which they do. I think that the "About" page should be added to this set of non-`800px` wide pages:

# Before

Screen Shot 2020-09-29 at 19 24 08


# After

Screen Shot 2020-09-29 at 19 23 40

It takes a whole lot of vertical scrolling for people to browse through these 3 at a time.

The `guide/cheatsheet` should also not be restricted to `800px` in width as this causes the code examples to be hard to read and adds quite a few horizontal scroll bars where content gets hidden.

The "Introductions to the Angular Docs" page needs some layout tweaks to work with this new layout (and has needed them for some time on smaller devices):

Screen Shot 2020-09-29 at 19 29 45

Other than that, I really like these changes and think that they make the docs a lot more comfortable to read on desktop devices.

**Comment by gkalpak** [comment]
Superceded by #40427. Let's continue the discussion there.
